servers/zts/src/main/java/com/yahoo/athenz/zts/store/DataStore.java

```diff
@@ -953,6 +969,11 @@ void deleteDomain(final String domainName) {
         changeLogStore.removeLocalDomain(domainName);
     }
 
+    private void processDeleteRolesRequireRoleCert(String domainName) {
+        List<String> rolesToDelete = requireRoleCertCache.asMap().keySet().stream().filter(key -> key.startsWith(domainName)).collect(Collectors.toList());
```

Comment:
this kind of logic with starts with will not work when you  have domains that include the same prefix - e.g. yahoo and yahoosports - if you delete yahoo then you'll also match yahoosports as well. In this cases you want to add the domain separator "yahoo." and then do the startsWith search

Reply:
Please see the latest changes (logic in cleanRequireRoleCertCache in DataStore.java).